File: Assets/_Scripts/States/Player/PlayerNormalState.cs
using Styx.Entities.PlayerModule;

namespace Styx.States
{
    public class PlayerNormalState : PlayerSkatingState
    {
        public override void Enter(PlayerState playerState)
        {
            base.Enter(playerState);
        }

        public override void Execute(PlayerState playerState)
        {
            base.Execute(playerState);
            playerState.Player.Inputs.DashKeyDown.Execute();

            if (playerState.Player.Inputs.DashKeyDown.IsPressed)
                playerState.Reduce(Message.UpdateToDash);
        }

        public override void FixedExecute(PlayerState playerState)
        {
            base.FixedExecute(playerState);
        }

        public override void Exit(PlayerState playerState)
        {
            base.Exit(playerState);
        }
    }
}

File: Assembly-FSharp/Assembly-FSharp/States/Player/PlayerNormalState.fs
namespace Styx.States.Player

open Styx.Entities.PlayerModule
open RageCure.StateUtils

type PlayerNormalState () =
    inherit PlayerSkatingState ()

    interface IState<Player> with

        member this.Enter (player: Player) =
            base.Enter(player)

        member this.Execute (player: Player) =
            base.Execute(player)

        member this.FixedExecute (player: Player) =
            base.FixedExecute(player)

        member this.Exit (player: Player) =
            base.Exit(player)